Mathlib/FieldTheory/SeparableClosure.lean

```diff
@@ -0,0 +1,374 @@
+  closure inside the algebraic closure.
+
+- `Field.sepDegree F E`: the (infinite) separable degree $[E:F]_s$ of an algebraic extension
+  `E / F` of fields, defined to be the degree of `separableClosure F E / F`. Later we will show
+  that (`Field.finSepDegree_eq`, not in this file), if `Field.Emb F E` is finite, then this
+  coincides with `Field.finSepDegree F E`.
+
+- `Field.insepDegree F E`: the (infinite) inseparable degree $[E:F]_i$ of an algebraic extension
+  `E / F` of fields, defined to be the degree of `E / separableClosure F E`.
+
+- `Field.finInsepDegree F E`: the (finite) inseparable degree $[E:F]_i$ of an algebraic extension
+  `E / F` of fields, defined to be the degree of `E / separableClosure F E` as a natural number.
+  It is zero if such field extension is not finite.
+
+## Main results
+
+- `le_separableClosure_iff`: an intermediate field of `E / F` is contained in the (relative)
+  separable closure of `E / F` if and only if it is separable over `F`.
+
+- `separableClosure.normalClosure_eq_self`: the normal closure of the (relative) separable
+  closure of `E / F` is equal to itself.
+
+- `separableClosure.isGalois`: the (relative) separable closure of a normal extension is Galois
+  (namely, normal and separable).
+
+- `separableClosure.isSepClosure`: the (relative) separable closure of a separably closed extension
+  is a separable closure of the base field.
+
+- `IntermediateField.isSeparable_adjoin_iff_separable`: `F(S) / F` is a separable extension if and
+  only if all elements of `S` are separable elements.
+
+- `separableClosure.eq_top_iff`: the (relative) separable closure of `E / F` is equal to `E`
+  if and only if `E / F` is separable.
+
+## Tags
+
+separable degree, degree, separable closure
+
+-/
+
+open scoped Classical Polynomial
+
+open FiniteDimensional Polynomial IntermediateField Field
+
+noncomputable section
+
+universe u v w
+
+variable (F : Type u) (E : Type v) [Field F] [Field E] [Algebra F E]
+
+variable (K : Type w) [Field K] [Algebra F K]
+
+section separableClosure
+
+/-- The (relative) separable closure of `E / F`, or called maximal separable subextension
+of `E / F`, is defined to be the intermediate field of `E / F` consisting of all separable
+elements. The previous results prove that these elements are closed under field operations. -/
+def separableClosure : IntermediateField F E where
+  carrier := {x | (minpoly F x).Separable}
+  mul_mem' := separable_mul
+  one_mem' := map_one (algebraMap F E) ▸ separable_algebraMap E 1
+  add_mem' := separable_add
+  zero_mem' := map_zero (algebraMap F E) ▸ separable_algebraMap E 0
+  algebraMap_mem' := separable_algebraMap E
+  inv_mem' := separable_inv
+
+/-- An element is contained in the (relative) separable closure of `E / F` if and only if
+it is a separable element. -/
+theorem mem_separableClosure_iff {x : E} :
+    x ∈ separableClosure F E ↔ (minpoly F x).Separable := Iff.rfl
```

Comment:
I think F and E need not be explicit. You might try to optimize other ones a bit too.